Azure/azure-cli
# [Storage] Fix #17858: `az storage blob upload`: make --name optional

**Description**

Previously, the --name parameter is required even when the file path has been passed. For customers' convenience, it can be optional. Fix #17858

**Testing Guide**

Previously -n is required
`az storage blob upload -f {file-path} -c {container-name} -n {blob-name}`
Now the name would be inferred from the file-path
`az storage blob upload -f {file-path} -c {container-name}`

---

This checklist is used to make sure that common guidelines for a pull request are followed.
